### DISCUSSION TITLE
How can I set up a key combination to toggle specific apps?

### DISCUSSION THREAD

**Discussion by sanity** [i3/i3#4759] [Q&A] [answered]
I try to divide my workspaces by task, for example assigning specific workspaces by project.

There are some apps that belong on specific workspaces because they're specific to particular tasks, but some that need to be used across tasks like email or IM clients.

I'd like to set up i3 such that a particular key combination will toggle a specific floating app on whichever workspace I'm on. For example, I might set up $mod+Alt+w to show WhatsApp on whatever workspace I'm currently on, and hit again to hide. Ideally, I'd like it to launch the app if it isn't already running.

Does anyone know of any recipes to do something like this?

**Comment by Airblader** [comment, marked as answer]
> For example, I might set up $mod+Alt+w to show WhatsApp on whatever workspace I'm currently on, and hit again to hide. 

This sounds like a use case for i3's scratchpad, maybe look into that in the userguide.

> Ideally, I'd like it to launch the app if it isn't already running.

i3 cannot do this for you because we don't know the relation between a window and the application that created it, but if you search for this ("run or focus", "run or raise") you'll find solutions in the shape of scripts.

**Comment by sanity** [comment]
Here was my solution (in this case for Slack):

```
exec --no-startup-id /snap/bin/slack
for_window [title="^Slack"] move scratchpad
bindsym $mod+Mod1+k [title="^Slack"] scratchpad show
```
